File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.py
N = int(input())

for _ in range(N):
    OX = input()
    score = 0
    extra = 0
    for i in range(len(OX)):
        if OX[i] == "O":
            if i != 0:
                if OX[i - 1] == "O":
                    extra += 1
                    score += extra
            score += 1
        else:
            extra = 0
    print(score)

File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		for (int i = 0; i < N; i++) {
			int score = 0;
			int extra = 0;
			String OX = sc.next();
			String[] listOX = OX.split("");
			for (int j = 0; j < listOX.length; j++) {
				if (listOX[j].equals("O")) {
					if (j != 0) {
						if (listOX[j - 1].equals("O")) {
							extra++;
							score += extra;
						}
					}
					score++;
				} else {
					extra = 0;
				}
			}
			System.out.println(score);
		}
		sc.close();
	}
}